cmd/pctl/search.go

```diff
@@ -25,42 +25,38 @@ func searchCmd() *cli.Command {
 				Value:       "table",
 				Usage:       "Output format. json|table",
 			},
+			&cli.BoolFlag{
+				Name:    "all",
+				Aliases: []string{"a"},
+				Usage:   "Search all available profiles",
+			},
 		},
 		Action: func(c *cli.Context) error {
-			searchName, catalogClient, err := parseArgs(c)
-			if err != nil {
-				_ = cli.ShowCommandHelp(c, "search")
-				return err
-			}
-
-			profiles, err := catalog.Search(catalogClient, searchName)
-			if err != nil {
-				return err
-			}
-			outFormat := c.String("output")
-			if outFormat == "table" {
-				if len(profiles) == 0 {
-					fmt.Printf("No profiles found matching: '%s'\n", searchName)
-					return nil
+			var profiles []profilesv1.ProfileCatalogEntry
+			if c.Bool("all") {
+				catalogClient, err := getCatalogClient(c)
+				if err != nil {
+					_ = cli.ShowCommandHelp(c, "search")
+					return err
+				}
+				profiles, err = catalog.Search(catalogClient, "")
+				if err != nil {
+					return err
+				}
+			} else {
+				searchName, catalogClient, err := parseArgs(c)
+				if err != nil {
+					_ = cli.ShowCommandHelp(c, "search")
+					return err
+				}
+				profiles, err = catalog.Search(catalogClient, searchName)
+				if err != nil {
+					return err
 				}
 			}
```

Comment:
could this be simplified and dried a bit? (not tested, btw, could be wrong)
```suggestion
      searchName, catalogClient, err := parseArgs(c)
      if err != nil {
        _ = cli.ShowCommandHelp(c, "search")
        return err
      }
      if c.Bool("all") {
        searchName = ""
      }
      profiles, err = catalog.Search(catalogClient, searchName)
      if err != nil {
        return err
      }
```